resources/testharness.js

```diff
@@ -3395,6 +3486,51 @@ policies and contribution forms [3].
             return '';
         }
 
+        var asserts_run_by_test = new Map();
```

Comment:
I wonder if it would be nicer to associate asserts with a Test right from the start? (I.e. instead of pushing them into a separate array). Right now every user of completion callback will have to write code to turn the entire assert array into a `test--> assert list` map.

The only question might be what to do in `set_assert` when `current_test` is `null`, which here is neatly handled because Map allows that. Of course maybe its a good chance to raise a harness error and be like "y'all missed a step call!"

Reply:
I didn't do that because in theory the ordering of asserts between tests could be significant. We don't actually expose that at the moment, but we could. We could store that in some other way of course e.g. by giving each assert an index. idk if you think that's valuable enough to retain the more complex data model.

Reply:
This is probably fine then. If we ever decide otherwise there's a fairly easy deprecation path (add to `Test` as well, then later remove the third argument)